aws-logging-cloudwatch/src/androidTest/kotlin/com/amplifyframework/logging/cloudwatch/db/CloudWatchLoggingDatabaseInstrumentationTest.kt

```diff
@@ -0,0 +1,155 @@
+        assertEquals("Customer Obsession", savedLogs[0].message)
+        assertEquals(testTimestamp2, savedLogs[1].timestamp)
+        loggingDbClass.clearDatabase()
+    }
+
+    /**
+     * This test verifies if the CloudWatch Logs are retrieved
+     */
+    @Test
+    fun queryAllEvents_Successfully_Retrieves_Every_Log() = runTest {
+        assertEquals(0, loggingDbClass.queryAllEvents().size)
+        loggingDbClass.saveLogEvent(testCloudWatchLogEvent1)
+        assertEquals(1, loggingDbClass.queryAllEvents().size)
+        loggingDbClass.saveLogEvent(testCloudWatchLogEvent3)
+
+        val savedLogs = loggingDbClass.queryAllEvents()
+        assertEquals("Bias for Action", savedLogs[savedLogs.size -1].message)
+        loggingDbClass.clearDatabase()
+    }
+
+    /**
+     * This test verifies if the CloudWatch Logs are correctly deleted by passing a list of one id
+     * to bulkDelete() method
+     */
+    @Test
+    fun bulkDelete_Removes_Only_One_Item() = runTest {
+        loggingDbClass.saveLogEvent(testCloudWatchLogEvent1)
+        loggingDbClass.saveLogEvent(testCloudWatchLogEvent2)
+        loggingDbClass.saveLogEvent(testCloudWatchLogEvent3)
+        loggingDbClass.saveLogEvent(testCloudWatchLogEvent4)
+        var deleteTargetLogs: MutableList<Long> = loggingDbClass.queryAllEvents().map { it.id } .toMutableList()
+        deleteTargetLogs = deleteTargetLogs.subList(0,1)
+
+        loggingDbClass.bulkDelete(deleteTargetLogs)
+
+        val savedLogs = loggingDbClass.queryAllEvents()
+
+        assertEquals(3, savedLogs.size)
+        assertEquals("Ownership", savedLogs[0].message)
+        loggingDbClass.clearDatabase()
+    }
+
+    /**
+     * This test verifies if the Cloudwatch Logs are correctly deleted by passing a list of two ids
+     * to bulkDelete() method
+     */
+    @Test
+    fun bulkDelete_Removes_With_List_Containing_Two_Ids() = runTest {
+        loggingDbClass.saveLogEvent(testCloudWatchLogEvent1)
+        loggingDbClass.saveLogEvent(testCloudWatchLogEvent2)
+        loggingDbClass.saveLogEvent(testCloudWatchLogEvent3)
+        loggingDbClass.saveLogEvent(testCloudWatchLogEvent4)
+        val deleteTargetLogs: MutableList<Long> = loggingDbClass.queryAllEvents().map { it.id } .toMutableList()
+        deleteTargetLogs.removeAt(1)
+        deleteTargetLogs.removeAt(2)
+
+        loggingDbClass.bulkDelete(deleteTargetLogs)
+
+        val savedLogs = loggingDbClass.queryAllEvents()
+
+        assertEquals(2, savedLogs.size)
+        assertEquals("Frugality", savedLogs[1].message)
+        loggingDbClass.clearDatabase()
+    }
+
+    /**
+     * This test verifies if the Database is cleared after clearDatabase() method is called
+     */
+    @Test
+    fun clearDatabase_Wipes_All_Logs_Out() = runTest {
+        assertEquals(0, loggingDbClass.queryAllEvents().size)
+        loggingDbClass.saveLogEvent(testCloudWatchLogEvent1)
+        assertEquals(1, loggingDbClass.queryAllEvents().size)
```

Comment:
These assertions are a bit superfluous since other tests already verify `saveLogEvent` works

Reply:
What I thought when writing these lines was that I should ensure the database is empty at start, and some random data that should not exist after `clearDatabase()` call is correctly removed. Seems it was redundant 🤔 . Removed!